common/src/dyn_regs.C

```diff
@@ -160,41 +182,83 @@ std::string MachRegister::name() const {
+                    ret = "SGPR["+std::to_string(base_val)+":"+std::to_string(base_val+1)+"]";
                     break;
                 case amdgpu_vega::VGPR_VEC2:
                     base_val -=  (amdgpu_vega::vgpr_vec2_0&0xff);
-                    ret = std::string("VGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+1)+"]");
+                    ret = "VGPR["+std::to_string(base_val)+":"+std::to_string(base_val+1)+"]";
                     break;
                 case amdgpu_vega::SGPR_VEC4:
                     base_val -=  (amdgpu_vega::sgpr_vec4_0&0xff);
-                    ret = std::string("SGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+3)+"]");
+                    ret = "SGPR["+std::to_string(base_val)+":"+std::to_string(base_val+3)+"]";
                     break;
                 case amdgpu_vega::VGPR_VEC4:
                     base_val -=  (amdgpu_vega::vgpr_vec4_0&0xff);
-                    ret = std::string("VGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+3)+"]");
+                    ret = "VGPR["+std::to_string(base_val)+":"+std::to_string(base_val+3)+"]";
                     break;
                 case amdgpu_vega::SGPR_VEC8:
                     base_val -=  (amdgpu_vega::sgpr_vec8_0&0xff);
-                    ret = std::string("SGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+7)+"]");
+                    ret = "SGPR["+std::to_string(base_val)+":"+std::to_string(base_val+7)+"]";
                     break;
                 case amdgpu_vega::VGPR_VEC8:
                     base_val -=  (amdgpu_vega::vgpr_vec8_0&0xff);
-                    ret = std::string("VGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+7)+"]");
+                    ret = "VGPR["+std::to_string(base_val)+":"+std::to_string(base_val+7)+"]";
                     break;
                 case amdgpu_vega::SGPR_VEC16:
                     base_val -=  (amdgpu_vega::sgpr_vec16_0&0xff);
-                    ret = std::string("SGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+15)+"]");
+                    ret = "SGPR["+std::to_string(base_val)+":"+std::to_string(base_val+15)+"]";
                     break;
                 case amdgpu_vega::VGPR_VEC16:
                     base_val -=  (amdgpu_vega::vgpr_vec16_0&0xff);
-                    ret = std::string("VGPR["+std::to_string(base_val)+"-"+std::to_string(base_val+15)+"]");
+                    ret = "VGPR["+std::to_string(base_val)+":"+std::to_string(base_val+15)+"]";
+                    break;
+                default:
+                    ret = iter->second;
                     break;
+            }
+            return ret;
+        }else if(getArchitecture() == Arch_amdgpu_cdna2){
+            signed int category = reg & 0x00ff0000;
+            signed int base_val = reg & 0x000000ff;
+            switch(category){
+                /*case amdgpu_cdna2::SGPR_VEC2:
+                  base_val -=  (amdgpu_cdna2::sgpr_vec2_0&0xff);
```

Comment:
Let's remove this commented-out code, if it's not needed.

Reply:
Yeah Ill clean these up.